### DISCUSSION TITLE
Is it possible to callback (a script - or await command) from a kit-divContainer ?

### DISCUSSION THREAD

**Discussion by wisskirchenj** [johnlindquist/kit#1106] [Q&A]
Hi,
I am Jürgen and first of all, sorry for a probably dumb question, but I just downloaded scriptkit today..
The first useful script I wrote just searches for some substring over all my Java-projects (locally) and greps code snippets, that contain the search string with some surrounding lines, highlights the search string and sets the file path as heading over all.
So I basically use zx to do some elaborate bash find -exec chain and parse the result via typescript into template strings. Looks like this:
<img width="1075" alt="Screenshot 2023-02-06 at 22 50 40" src="https://user-images.githubusercontent.com/99090177/217097444-5f14e949-e406-42b1-b3d6-da0de5925a0c.png">
So far so good: My next step - and this is my question - would be, to make each Filepath-Heading (e.g the files containing a search result) a link. If I then click on such a link, I want my intelliJ idea have the associated project open - which basically is just another zx command, like "idea <project-path>"..
Is this achievable - or is the div-Container the wrong tool to have this workflow running ?

Hope this was understandable and I would love to get some advice. Thks a lot!
Best regards, Jürgen

**Comment by wisskirchenj** [comment]
OK - I got it working (nearly) as I want - though there is some room for improvement ..
=> I extract filenames into a widget with buttons for each project, put a click listener on the widget. And on click I open my IDE with the project of the clocked file.. some string parsing into commands - but that's it
-> Many thanks to script kit - cool to be able to accomplish this on my first day, which shows the power of this :-)

**Reply by johnlindquist** [reply]
@wisskirchenj Hi! 👋

I would use `arg` with previews. `preview` can be an async function which loads any content into the right side panel:

```js
// Name: Artist Search

import "@johnlindquist/kit"

// If you're not getting any image results, run:
// cd ~/.kenv && npm i googlethis@latest

let google = await npm("googlethis")

let artists = [
  "Leonardo da Vinci",
  "Frida Kahlo",
  "Michelangelo",
  "Georgia O'Keeffe",
  "Raphael",
  "Rembrandt",
  "Mary Cassatt",
  "Vermeer",
  "Van Gogh",
  "Monet",
]

let options = {
  page: 0,
  safe: false,
  additional_params: {
    // add additional parameters here, see https://moz.com/blog/the-ultimate-guide-to-the-google-search-parameters and https://www.seoquake.com/blog/google-search-param/
    hl: "en",
  },
}

let selectedArtist = await arg(
  "Select an Artist",
  async () => {
    return artists.map(artist => {
      return {
        name: artist,
        value: artist,

        preview: async () => {
          let response = await google.image(artist, options)
          let images = response
            .map(i => i?.preview?.url || "")
            .map(
              src =>
                `<img class="object-contain w-24 h-24" src="${src}"/>`
            )
            .join("")

          return `<div class="container flex flex-wrap">${images}</div>`
        },
      }
    })
  }
)
```
![CleanShot 2023-02-06 at 17 42 40](https://user-images.githubusercontent.com/36073/217119221-caa5b79a-0810-448a-a352-186ae33a11f7.png)


If you show me the grep command you're using, I can make a better demo to explain it:

**Reply by wisskirchenj** [reply]
@johnlindquist: Thank you so much for the heartly welcome and most informative reply of yours :smile: :heart:
You are right, the preview is a good alternative for my use case! And all the better it is good to know, that this way exists.

Having said that, for my personal needs, the 3-step workflow I implemented now, i.e.
1. input search parameters
2. display them nicely as html in a scrollable div-container to walk-through
3. advance by pressing enter to a widget-dialog that lets me open the IDE project for further investigation and code copying ...
is just perfect - so I may stick with that 😄. 

Here it is on GitHub - in case you are interested: [Juergen's find-code kit-script](https://github.com/wisskirchenj/kit-scripts/blob/main/scripts/find-code.ts)

Two more **questions** - if I may:

1. Is there a way to have the script (that I start by shortcut) come up with some fixed (bigger width)? and if yes, how to specify...
2. Is there a way to "send keystrokes" to another window / app? For my code-finding use case e.g. I now open the IntelliJ Idea project to a selected search result, but I cannot open the file containing the result in IDE's editor. This were possible, if I'd sent some 'Cmd O' for file search and keystrokes of file name to the just opened window... Maybe it's too greedy to ask 😊, but you might have considered it already - or at least you seem to be most interested in future ideas !

BTW: As it might interest you: I found this place from an enthusiasting UTube of **Matt Pocock**   - who does amazing stuff with Kit :+1:
I am thankful to him for showing me, but especially to you for providing Script Kit. THX!!!

**Reply by johnlindquist** [reply]
@wisskirchenj 

1. Try this for a wider prompt:

```js
await arg({
  placeholder: "I should be wider",
  width: 1200,
}, ["One", "Two"])
```

> Note: There's a bug if you have multiple prompts in the same script with different widths that I need to sort out

2. Try this:

```js
await arg("After this prompt, it will send a cmd+o")

await hide()
await keyboard.pressKey(Key.LeftSuper, Key.O)
await keyboard.releaseKey(Key.LeftSuper, Key.O)
```

> Note: Kit.app will need "accessibility permissions"

**Reply by wisskirchenj** [reply]
Thks a lot:
@1: Yeah - makes sense ... should have figured that out myself 😊 -> works well...
@2: that is actually kind of hidden in the guides yet :-) -> will try that, setSelectedText might help too... will see
